### PR TITLE
DO NOT MERGE - progress report

### PR DESCRIPTION
**Pull request by toasterbrain**
is there a better way to do this? let y'all see my code that is.

basically im posting what i have in the hope that someone can tell me what is wrong with it.
going from the index loads but hits me with the default whitelabel error when i try to look at the list of semesters. is this because that list is empty?

i will be posting new errors as a find them in the thread i started in the slack channel.

### CODE REVIEW

File: src/main/java/unbanner/Semester.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.util.List;
+
+public class Semester implements Storable {
+
+
+    @Id
+    public ObjectId id;
+
+    String season;
```
Comment (mkbhattarai) [inline src/main/java/unbanner/Semester.java]: This can be enum for SPRING, FALL, SUMMER

File: src/main/java/unbanner/Semester.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    @DBRef(lazy = true)
+    @Getter
+    @Setter
+    public List<Section> sections;
+
+
+
+    public Semester(String season, int year) {
+        this.season = season;
+        this.year = year;
```
Comment (mkbhattarai) [inline src/main/java/unbanner/Semester.java]: can check make sure year is not negative

File: src/main/java/unbanner/Section.java [hunk 3]
```diff
@@ -32,6 +32,11 @@
   @DBRef(lazy = true)
   public Room room;
   @DBRef(lazy = true)
+  public Semester semester;
+<<<<<<< HEAD
```
Comment (cblanto7) [inline src/main/java/unbanner/Section.java]: merge conflict that needs to be resolved

File: src/main/resources/templates/semester.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org"
+      xmlns:layout="http://www.ultraw.net.nz/thymeleaf/layout"
+      layout:decorator="layout">
+<head>
+    <meta charset="UTF-8">
```
Comment (cblanto7) [inline src/main/resources/templates/semester.html]: this is why you are getting a whitelabel error, this line should be
`<meta charset="UTF-8"/>`

File: src/main/resources/templates/semesters.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org"
+      xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
+      layout:decorator="layout">
+<head>
+    <meta charset="UTF-8">
```
Comment (cblanto7) [inline src/main/resources/templates/semesters.html]: again this line should be
`<meta charset="UTF-8"/>`

File: src/main/resources/templates/semesters.html [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            </tr>
+            <tr th:each="semester : ${semesters}">
+                <td th:text="${semester.season}"></td>
+                <td th:text="${semester.year}"></td>
+                <td><a class="badge badge-info"
+                       th:href="@{/semester/{id}(id=${semester.id})}">view
+                </a></td>
+            </tr>
+            <tr>
+                <a class="badge badge-primary" th:href=
```
Comment (cblanto7) [inline src/main/resources/templates/semesters.html]: change this line to 
`<a class="badge badge-primary" href="/semesters/new">Create New Semester</a>`

**Comment by cblanto7** [comment]
I have no idea what travis-ci is doing but it seems to be having issues with this pull request for some reason

**Comment by toasterbrain** [comment]
this one appears to be broken as casey pointed out. im making a new one

**Pull request closed without merging**